Import 

In [2]:
import pandas as pd

In [3]:
with open('/Users/mingo/Documents/BI/MP1/champion_info.json') as f:
 data = json.load(f)
#df = pd.read_json(f)

In [4]:
# Extract the champion data part
champ_data = data['data']

In [5]:
# Now make a DataFrame
df = pd.DataFrame.from_dict(champ_data, orient='index')

In [6]:
# Hvilke kolonner findes der?
df.columns

Index(['title', 'id', 'key', 'name'], dtype='object')

In [7]:
# Print dataframmet ud
df

title   id          key          name
1                 the Dark Child    1        Annie         Annie
2                  the Berserker    2         Olaf          Olaf
3                   the Colossus    3        Galio         Galio
4                the Card Master    4  TwistedFate  Twisted Fate
5       the Seneschal of Demacia    5      XinZhao      Xin Zhao
..                           ...  ...          ...           ...
429       the Spear of Vengeance  429      Kalista       Kalista
432      the Wandering Caretaker  432         Bard          Bard
497                  The Charmer  497        Rakan         Rakan
498                    the Rebel  498        Xayah         Xayah
516  The Fire below the Mountain  516         Ornn          Ornn

[138 rows x 4 columns]

Here you can see the printet dataframe.

In [9]:
# De første fem rækker
df.head()

title  id          key          name
1            the Dark Child   1        Annie         Annie
2             the Berserker   2         Olaf          Olaf
3              the Colossus   3        Galio         Galio
4           the Card Master   4  TwistedFate  Twisted Fate
5  the Seneschal of Demacia   5      XinZhao      Xin Zhao

here we just see the first 5 rows of the dataframe prinet

In [11]:
# De sidste fem rækker
df.tail()


title   id      key     name
429       the Spear of Vengeance  429  Kalista  Kalista
432      the Wandering Caretaker  432     Bard     Bard
497                  The Charmer  497    Rakan    Rakan
498                    the Rebel  498    Xayah    Xayah
516  The Fire below the Mountain  516     Ornn     Ornn

and here we just see the last five rows 

In [13]:
# Tilfældige fem rækker
df.sample(5)

title   id      key     name
67       the Night Hunter   67    Vayne    Vayne
43    the Enlightened One   43    Karma    Karma
202          the Virtuoso  202     Jhin     Jhin
76   the Bestial Huntress   76  Nidalee  Nidalee
154     the Secret Weapon  154      Zac      Zac

Here in the dataframe I just randomly select 5 rows of data

Data cleaning

In [16]:
#get an overview 
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 138 entries, 1 to 516
Data columns (total 4 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   title   138 non-null    object
 1   id      138 non-null    int64 
 2   key     138 non-null    object
 3   name    138 non-null    object
dtypes: int64(1), object(3)
memory usage: 5.4+ KB


With the info seen above we can conluded there is no data to clean, since there is 138 entries which matches with 138 non-null

In [18]:
# i have made this exmample of how i would clean up the datafram if it had any NaN values
#df = df.drop(['key'], axis=1)

In [19]:
# print the dataframe
df

title   id          key          name
1                 the Dark Child    1        Annie         Annie
2                  the Berserker    2         Olaf          Olaf
3                   the Colossus    3        Galio         Galio
4                the Card Master    4  TwistedFate  Twisted Fate
5       the Seneschal of Demacia    5      XinZhao      Xin Zhao
..                           ...  ...          ...           ...
429       the Spear of Vengeance  429      Kalista       Kalista
432      the Wandering Caretaker  432         Bard          Bard
497                  The Charmer  497        Rakan         Rakan
498                    the Rebel  498        Xayah         Xayah
516  The Fire below the Mountain  516         Ornn          Ornn

[138 rows x 4 columns]

now the dataframe is cleaned(if i had removed any values)

In [21]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 138 entries, 1 to 516
Data columns (total 4 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   title   138 non-null    object
 1   id      138 non-null    int64 
 2   key     138 non-null    object
 3   name    138 non-null    object
dtypes: int64(1), object(3)
memory usage: 5.4+ KB


we will compare the non-null value to the entries and as we can see it matched with non-null value. therefor we can conluded that the dataframe is clean now 